git-object/src/commit/write.rs

```diff
@@ -21,6 +22,38 @@ impl crate::WriteTo for Commit {
         out.write_all(&self.message)
     }
 
+    fn size(&self) -> usize {
+        let hashsize = self.tree.kind().len_in_hex();
+        b"tree".len()
```

Comment:
`#[rustfmt::skip]` would help to have more specialized formatting without `cargo fmt` exploding it into newlines.

However, I'd really love to see what the number literals are there for, to help maintenance and be a learning experience at the same time as it's basically a description of the serialized format then :).

Reply:
Yep, the numeric constants are really just the literal characters (spaces and newlines mostly) inserted by the `trusted_*` functions so e.g. `trusted_header_id(key, value)` is `key <space> hex <newline>` hence `key.len() + 1 + len_in_hex + 1`.

But I think I'll update this to add e.g. `#[inline] encode::trusted_header_id_size(key, value) -> usize` that way the non-trivial `size` implementations of Commit and Tag should be much easier to relate to the `write_to` implementation. I waffled a bit on doing it before pushing, but after your comments and the mess rustfmt makes of it it seems like a cleaner solution than `rustfmt::skip`.